tests/questionSharing.test.js

```diff
@@ -0,0 +1,257 @@
+      let result = UUID_REGEXP.exec(response.text());
+      exampleCourseSharingId = result ? result[0] : null;
+      assert(exampleCourseSharingId != null);
+    });
+
+    step('Create a sharing set', async () => {
+      const sharingUrl = sharingPageUrl(testCourseId);
+      let response = await helperClient.fetchCheerio(sharingUrl);
+      const token = response.$('#test_csrf_token').text();
+      await fetch(sharingUrl, {
+        method: 'POST',
+        headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+        body: new URLSearchParams({
+          __action: 'sharing_set_create',
+          __csrf_token: token,
+          sharing_set_name: sharingSetName,
+        }).toString(),
+      });
+
+      let sharingPage = await (await fetch(sharingPageUrl(exampleCourseId))).text();
+      assert(sharingPage.includes(exampleCourseSharingName));
+    });
+
+    // step('Attempt to create another sharing set with the same name', async () => {
+    //   // TODO ensure that the sharing set name you created only appears once on the page
+    // });
+
+    // step('Attempt to create a sharing set with an invalid name', async () => {
+
+    // });
+
+    step('Share sharing set with example course', async () => {
+      const sharingUrl = sharingPageUrl(testCourseId);
+      let response = await helperClient.fetchCheerio(sharingUrl);
+      const token = response.$('#test_csrf_token').text();
+      await fetch(sharingUrl, {
+        method: 'POST',
+        headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+        body: new URLSearchParams({
+          __action: 'course_sharing_set_add',
+          __csrf_token: token,
+          sharing_set_id: '1',
+          course_sharing_id: exampleCourseSharingId,
+        }).toString(),
+      }); // TODO: should this endpoint return an error if the sharing set passed in is not valid or doesn't belong to the course?
+
+      let sharingPage = await (await fetch(sharingPageUrl(testCourseId))).text();
+      assert(sharingPage.includes('XC 101'));
+    });
+
+    // step('Attempt to share sharing set with invalid course ID', async () => {
+
+    // });
+
+    // step('Attempt to create another sharing set with the same name', async () => {
+
+    // });
+
+    step('Add question "addNumbers" to sharing set', async () => {
+      // TODO: should this block of code be factored out to helperClient as a
+      // helper function for getting to the page of a question with a given qid?
+      // or does this code already exist somewhere and I am duplicating effort here?
+      const questionsUrl = `${baseUrl}/course/${testCourseId}/course_admin/questions`;
+      const questionsPage = await helperClient.fetchCheerio(questionsUrl);
+      const questionData = questionsPage.$('#questionsTable').attr('data-data');
+      const questions = JSON.parse(questionData);
+      const addNumbersInfo = questions.find((questionInfo) => questionInfo.qid === 'addNumbers');
+
```

Comment:
@nwalters512 these few lines are just grabbing the link to a particular question from the questions page. Is there are better way to do this? or is it already done for me an a helper that I'm missing? I mean, I guess I could just run a DB query to get the question id using the qid and that would be less lines of code than this.